tsl/test/sql/foreign_keys.sql

```diff
@@ -0,0 +1,306 @@
+ROLLBACK;
+
+-- should cascade
+BEGIN;
+DELETE FROM metrics WHERE time = '2020-01-01';
+SELECT * FROM event ORDER BY event;
+ROLLBACK;
+-- should cascade
+BEGIN;
+UPDATE metrics SET time = '2020-01-01 00:30:00' WHERE time = '2020-01-01';
+SELECT * FROM event ORDER BY event;
+ROLLBACK;
+
+SELECT conname, conrelid::regclass, confrelid::regclass, conparentid <> 0 AS parent FROM pg_constraint WHERE conrelid='event'::regclass ORDER BY oid;
+SELECT tgfoid::regproc, tgparentid <> 0 AS parent, tgisinternal, tgconstrrelid::regclass FROM pg_trigger WHERE tgconstrrelid='event'::regclass ORDER BY oid;
+
+DROP TABLE event;
+DROP TABLE metrics;
+
+-- test single column fk constraint from plain table to hypertable with constraint being added separately
+CREATE TABLE metrics(time timestamptz primary key, device text, value float);
+SELECT table_name FROM create_hypertable('metrics', 'time');
+
+INSERT INTO metrics(time, device, value) VALUES ('2020-01-01', 'd1', 1.0);
+
+CREATE TABLE event(time timestamptz, info text);
+
+ALTER TABLE event ADD CONSTRAINT event_time_fkey FOREIGN KEY (time) REFERENCES metrics(time) ON DELETE RESTRICT;
+
+-- should fail
+\set ON_ERROR_STOP 0
+INSERT INTO event(time, info) VALUES ('2020-01-02', 'info1');
+\set ON_ERROR_STOP 1
+-- should succeed
+INSERT INTO event(time, info) VALUES ('2020-01-01', 'info2');
+
+-- should fail
+\set ON_ERROR_STOP 0
+DELETE FROM metrics WHERE time = '2020-01-01';
+\set ON_ERROR_STOP 1
+
+SELECT conname, conrelid::regclass, confrelid::regclass, conparentid <> 0 AS parent FROM pg_constraint WHERE conrelid='event'::regclass ORDER BY oid;
+SELECT tgfoid::regproc, tgparentid <> 0 AS parent, tgisinternal, tgconstrrelid::regclass FROM pg_trigger WHERE tgconstrrelid='event'::regclass ORDER BY oid;
+
+DROP TABLE event;
+DROP TABLE metrics;
+
+-- test single column fk constraint from plain table to hypertable with constraint being added separately while data is present
+CREATE TABLE metrics(time timestamptz primary key, device text, value float);
+SELECT table_name FROM create_hypertable('metrics', 'time');
+
+INSERT INTO metrics(time, device, value) VALUES ('2020-01-01', 'd1', 1.0);
+
+CREATE TABLE event(time timestamptz, info text);
+INSERT INTO event(time, info) VALUES ('2020-01-01', 'info2');
+INSERT INTO event(time, info) VALUES ('2020-02-01', 'info1');
+
+-- should fail
+\set ON_ERROR_STOP 0
+ALTER TABLE event ADD CONSTRAINT event_time_fkey FOREIGN KEY (time) REFERENCES metrics(time) ON DELETE SET NULL;
+\set ON_ERROR_STOP 1
+
+INSERT INTO metrics(time, device, value) VALUES ('2020-02-01', 'd1', 1.0);
+ALTER TABLE event ADD CONSTRAINT event_time_fkey FOREIGN KEY (time) REFERENCES metrics(time) ON DELETE CASCADE;
+
+-- delete should cascade
+DELETE FROM metrics WHERE time = '2020-01-01';
+SELECT * FROM event;
+
+DROP TABLE event;
+DROP TABLE metrics;
+
+-- test single column fk constraint from hypertable to hypertable
+-- test multi-column fk constraint from plain table to hypertable
```

Comment:
Looks like this is still to do, and also would be good to have some tests with compressed hypertable.